streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -153,18 +155,20 @@ public void advanceTimeMs(final long advanceMs) {
     }
 
     /**
-     * Create a {@link ConsumerRecord} with the given topic name, key, value, and timestamp.
+     * Create a {@link ConsumerRecord} with the given topic name, key, value, headers, and timestamp.
      * Does not auto advance internally tracked time.
      *
      * @param topicName the topic name
      * @param key the record key
      * @param value the record value
+     * @param headers the record headers
      * @param timestampMs the record timestamp
      * @return the generated {@link ConsumerRecord}
      */
     public ConsumerRecord<byte[], byte[]> create(final String topicName,
                                                  final K key,
                                                  final V value,
+                                                 final Headers headers,
                                                  final long timestampMs) {
         Objects.requireNonNull(topicName, "topicName cannot be null.");
         final byte[] serializedKey = keySerializer.serialize(topicName, key);
```

Comment:
Should we add a check that `headers` is not `null`? (if yes, please add test, too)

Reply:
Agree, added